src/main/java/com/hubspot/jinjava/el/ext/JinjavaBeanELResolver.java

```diff
@@ -208,10 +227,19 @@ private static int pickMoreSpecificMethod(Method methodA, Method methodB) {
     return 1;
   }
 
-  private String validatePropertyName(Object property) {
+  private String validatePropertyName(ELContext context, Object property) {
     String propertyName = transformPropertyName(property);
 
-    if (RESTRICTED_PROPERTIES.contains(propertyName)) {
+    JinjavaInterpreter interpreter = (JinjavaInterpreter) context
+      .getELResolver()
+      .getValue(context, null, ExtendedParser.INTERPRETER);
+
+    Set<String> allRestrictedProperties = Sets.union(
+      interpreter.getConfig().getRestrictedProperties(),
+      DEFAULT_RESTRICTED_PROPERTIES
+    );
```

Comment:
My hunch is that it's faster to check if either set contains it rather than unionizing the sets.

Reply:
I changed it to a check instead of unionizing the sets.